app/renderer/js/preload.js

```diff
@@ -78,6 +82,10 @@ document.addEventListener('DOMContentLoaded', () => {
 			}
 		});
 	}
+	// Update user status after every 15s
+	setInterval(() => {
+		ipcRenderer.send('detect-presence');
+	}, 15000);
```

Comment:
Just to check, I assume this is code that runs only once if you have N organizations?

Reply:
I would have thought we'd just have a single copy of the setInterval() calling `electron.powerMonitor.querySystemIdleState`, and having that send IPC data to each Zulip app window, but maybe I'm missing something here.

Reply:
Each org has a different preload.js in their webview. So, this is called differently for each org. I couldn't figure out a way to do this only once and then send it to each window. Will look into it more.

Reply:
@vsvipul  to address this issue - let's move the the `setInterval` to main process so that it occurs once for N orgs and send a IPC data to the `preload.js` directly with `set-active` or `set-idle`. Can you try this out?

Reply:
@abhigyank I'll try it out. Thanks. This is what i was doing initially, but someone pointed out that the SetInterval should be placed in preload.js and not in the main.

Reply:
@abhigyank @timabbott  I have moved the logic to main.js and now it runs only once and sets the variable for all webviews. I hope this resolves the issue.

Reply:
The updated change on this looks good to me.